### ISSUE TITLE
Bad results with 3 state Markov Chain Neural Network

### ISSUE THREAD

**Issue by s-agapov** [aadya940/chainopy#28]
Hello. Interesting work, but I am having trouble reproducing your results.

The code from example notebook:
```
_markov_chain = MarkovChain(
        [[0.3, 0.5, 0.2], 
         [0.1, 0.8, 0.1], 
         [0.3, 0.3, 0.4]],

        states=["Rainy", "Sunny", "Cloudy"],
)

model = MarkovChainNeuralNetwork(_markov_chain, num_layers=5)
model.train_model(
                  num_samples=30_000, 
                  epochs=1000, 
                  learning_rate=0.1, 
                  momentum=0.9, 
                  verbose=False, 
                  patience=500, 
                  factor=0.5
                )
```

 Almost all time loss   > 0.7 and results too bad.
![image](https://github.com/user-attachments/assets/c1aabfda-8cf7-4800-b0a1-1e9c6d1ee7c3)

**Comment by aadya940** [comment]
Hey @s-agapov , Thank you for opening an issue!

Unfortunately, gradient descent is not deterministic in nature. Can you try running this a couple of times?
